src/shell.js

```diff
@@ -71,6 +71,27 @@ if (!ENVIRONMENT_IS_PTHREAD) PthreadWorkerInit = {};
 var currentScriptUrl = ENVIRONMENT_IS_WORKER ? undefined : document.currentScript.src;
 #endif
 
+if (typeof Module['locateFile'] !== 'function') {
+  if (Module['memoryInitializerPrefixURL'] || Module['pthreadMainPrefixURL'] || Module['cdInitializerPrefixURL'] || Module['filePackagePrefixURL']) {
+    throw new Error("Module['locateFile'] can't be used together with Module['*PrefixURL'] options; if you need to customize prefix, take a look at Module['scriptDirectory'] option instead");
+  }
+  // `/` should be present at the end if `Module['scriptDirectory']` is not empty
+  if (!Module['scriptDirectory']) {
+    if (ENVIRONMENT_IS_NODE) {
+      Module['scriptDirectory'] = __dirname + '/';
+    } else if (ENVIRONMENT_IS_WEB) {
+      Module['scriptDirectory'] = document.currentScript.src.split('/').slice(0, -1).join('/') + '/';
+    } else if (ENVIRONMENT_IS_WORKER) {
+      Module['scriptDirectory'] = self.location.href.split('/').slice(0, -1).join('/') + '/';
```

Comment:
I think if someone uses `<base>`, they explicitly want a different base path for any resources and that shouldn't get broken.